tests/unit/test_global_correction.py

```diff
@@ -106,3 +109,84 @@ def test_lhc_global_correct_methods(tmp_path, model_inj_beams, method):
 
     if method == "pinv":
         assert len(correction.index) > n_correctors  # unless by accident
+
+@pytest.mark.basic
+def test_update_response(tmp_path, model_inj_beams):
+    """ Tests if the response is updated. """
+    # create the accelerator instance
+    beam = model_inj_beams.beam
+    knob = "kqd.a78"#f'kq10.l1b{beam}'
+    accel_inst = manager.get_accelerator(model_inj_beams)
+    delta = tfs.TfsDataFrame(
+        3.34e-05,
+        index=[knob], 
+        columns=[DELTA]
+        )
+    optics_params = ['PHASEX', 'PHASEY']
+
+    # Get a response dict from the model
+    ref_resp_dict = response_madx.create_fullresponse(accel_inst, [knob])
+
+    corr_model_path = tmp_path / f"twiss_cor{EXT}"
+    same_model = _create_corrected_model(corr_model_path, [], accel_inst, False)
+    
+    # As orbit_dpp is not in the response, it should not be updated. First for response_madx
+    new_resp_dict = _update_response(
+        accel_inst=accel_inst,
+        corrected_elements=same_model,
+        optics_params=optics_params,
+        corr_files=[],
+        variable_categories=[knob],
+        update_dpp=False,
+        update_response="madx"
+)
+
+    # Check everything is the same, as model has not changed.
+    for key in ref_resp_dict.keys():
+        assert_frame_equal(ref_resp_dict[key], new_resp_dict[key])
+
+    corr_file = tmp_path / "corr.madx"
+    with open(corr_file, "w") as f:
+        f.write(f"{knob} = {knob}{delta.loc[knob, DELTA]:+.16e};")
+
+    corr_model = _create_corrected_model(corr_model_path, [corr_file], accel_inst, False)
+    assert not same_model.equals(corr_model)
+
+    # Now for response_madx with the model changed
```

Comment:
Actually, that the model has changed is irrelevant for the madx-response update. 

I think you should create the updated model only after the madx-response  test, as it belongs to the twiss-response update, which uses the data from the twiss (i.e. the "df_elements") to calcualte the response.